app/models/transformation_mapping/vm_migration_validator.rb

```diff
@@ -98,7 +99,7 @@ def vm_migration_status(vm)
     return VM_MIGRATED unless vm_as_resources.where(:status => ServiceResource::STATUS_COMPLETED).empty?
 
     # VM failed in previous migration
-    vm_as_resources.all? { |rsc| rsc.status == ServiceResource::STATUS_FAILED } ? VM_VALID : VM_IN_OTHER_PLAN
+    vm_as_resources.all? { |rsc| rsc.status == ServiceResource::STATUS_FAILED || rsc.service_template_id.to_s == @service_template_id  } ? VM_VALID : VM_IN_OTHER_PLAN
```

Comment:
Why is `@service_template_id` a string?

Reply:
The UI has `service_template_id ` info available to it as a string (extracted from a JSON), so it sends that as is to the backend.

Usually the interaction between the UI and the backend when it comes to id's is always in the string format.

Reply:
I know about the JSON limitation with bigint, but once we're in a model, I would expect `anything_id` to be a bigint.

Reply:
Ok, in that case I would have to apply `.to_i` to `service_template_id` in https://github.com/ManageIQ/manageiq/blob/c3a331db747cb4de1430a6e6e5a14f59db8274bf/app/models/transformation_mapping/vm_migration_validator.rb#L16
Does that work?

Reply:
@bdunne I have made the above change in the last commit c7dc8ae
Does that look OK?